File: array_shift/array_shift.py

# ATOMIC_FUNCTION: array_shift(array)
# TAGS: list
# LANG:  python
# FUNCTION: array_shift(array)
def array_shift(array):
    if len(array)>1:
        return array[1:]
    else:
        return None

start_val_1 = ["orange", "banana", "apple", "raspberry"]
expected_end_val = ["banana", "apple", "raspberry"]
end_val = array_shift(start_val_1)

# TEST
if(end_val == expected_end_val):
    print("PASS")
else:
    print("FAIL")

File: array_shift/array_shift.php
<?php
// ATOMIC_FUNCTION: array_shift($a)
// TAGS: array
// LANG:  php
// FUNCTION: array_shift($a)

$start_val_1 = array("orange", "banana", "apple", "raspberry");
$expected_end_val = array("banana", "apple", "raspberry");
array_shift($start_val_1);
$end_val = $start_val_1;

// TEST
if($end_val == $expected_end_val) {
	echo "PASS";
} else {
	echo "FAIL";
}

?>
